### PR TITLE
[WIP] feat: CAMT.053 import

### PR DESCRIPTION
**Pull request by mvdkleijn**
This is a draft PR.

This PR introduces some initial code to import CAMT.053 based XML files into Actual.

Feedback is very much welcome. If you are a user and want to test this code, feel free but do NOT do so on your production system. Anyone that wants to contribute testing code is also more than welcome.

Status:

- [x] Check CAMT.053 specs
- [x] Make xml files show up in import file selection
- [x] Read transactions from xml file into Actual
- [x] Test using real CAMT.053 exports
- [ ] Fix bugs
- [ ] Write basic tests
- [ ] Double check everything against CAMT.053 specs
- [ ] Support multiple file import?

### CODE REVIEW

**Comment by mvdkleijn** [comment]
This is a DRAFT PR for issue #513

File: packages/loot-core/src/server/accounts/parse-file.js [hunk 1]
```diff
@@ -57,6 +59,106 @@ async function parseCSV(filepath, options = {}) {
...
+    let tmp = {};
+    tmp.date = element.ValDt.Dt._text;
+    
+    if (element.CdtDbtInd._text == "DBIT") {
+      tmp.amount = -looselyParseAmount(element.Amt._text);
+    } else {
+      tmp.amount = looselyParseAmount(element.Amt._text);
+    }
+    
+    if (typeof element.NtryDtls === 'undefined' || element.NtryDtls === null) {
```
Comment (j-f1) [inline packages/loot-core/src/server/accounts/parse-file.js:75]: Those are some excellent property names they give you :) nit: you can use `element.NtryDtls == null` to check for both

File: package.json [hunk 2]
```diff
@@ -51,5 +51,8 @@
     "@babel/helper-plugin-utils": "^7.14.5",
     "react-error-overlay": "6.0.9"
   },
-  "packageManager": "yarn@3.2.0"
+  "packageManager": "yarn@3.2.0",
+  "dependencies": {
+    "xml-js": "^1.6.11"
```
Comment (j-f1) [inline package.json:56]: Can you move this to the appropriate subpackage?

File: packages/loot-core/src/server/accounts/parse-file.js [hunk 3]
```diff
@@ -57,6 +59,106 @@ async function parseCSV(filepath, options = {}) {
...
+    errors.push({
+      message: 'Failed parsing: ' + err.message,
+      internal: err.message
+    });
+    return { errors, transactions: [] };
+  }
+
+  return {
+    errors,
+    transactions: data.map(trans => ({
```
Comment (j-f1) [inline packages/loot-core/src/server/accounts/parse-file.js:151]: It seems like this `map` can be removed? It doesn’t look like it’s doing anything.

**Review by j-f1 — COMMENTED** [review]
This is super cool! I left a few comments but feel free to ignore them as you continue working :)

**Comment by MatissJanis** [comment]
👋 Hey! Just checking in if you're still planing to work on this?

The PR has been inactive for 3 weeks now. Generally I try closing off inactive PRs after a month of inactivity, but please let me know if you wish to continue working here and I'll keep it open for you.

**Comment by mvdkleijn** [comment]
Hey, yeah sorry. I am planning to work on this more. I just have very scattered time and three weeks is over before I realize. Hopefully I should have some time this weekend.

**Comment by MatissJanis** [comment]
👋 I really appreciate all the work you did here so far, but the branch seems to have fallen behind quite a lot in history. And there hasn't been any recent development here.

So for now I'll close this. But when/if you resume working on this - please feel free to open a new PR. We'll be more than happy to get this in `master` branch :)

**Comment by sNiXx** [comment]
@mvdkleijn Because I would love to use this as well: Are you still planning to work on this PR?

**Comment by mvdkleijn** [comment]
> @mvdkleijn Because I would love to use this as well: Are you still planning to work on this PR?

No sorry. After the PR got closed because I didn't have time to finish it, I decided to go a different route as this software didn't quite fit my bill.

**Pull request closed without merging**